### PR TITLE
feat: add custom channel in github provider

### PR DESCRIPTION
**Pull request by beyondkmp**
fix https://github.com/electron-userland/electron-builder/issues/8388

### CODE REVIEW

**Review by mmaietta — APPROVED** [review]

File: packages/electron-updater/src/providers/GitHubProvider.ts [hunk 1]
```diff
@@ -136,7 +141,7 @@ export class GitHubProvider extends BaseGitHubProvider<GithubUpdateInfo> {
     }
 
     try {
-      const channel = this.updater.allowPrerelease ? this.getCustomChannelName(String(semver.prerelease(tag)?.[0] || "latest")) : this.getDefaultChannelName()
+      const channel = this.updater.allowPrerelease ? this.getCustomChannelName(String(semver.prerelease(tag)?.[0] || "latest")) : this.channel
```
Comment (benbucksch) [inline packages/electron-updater/src/providers/GitHubProvider.ts:144]:
```suggestion
      const channel = this.updater.allowPrerelease ? this.getCustomChannelName(String(semver.prerelease(tag)?.[0] || this.channel)) : this.channel
```

Comment (benbucksch) [reply packages/electron-updater/src/providers/GitHubProvider.ts:144]: Even if we deploy 0.4.0alpha1, we still want it to use the channel name that we set in the config file. The `latest` will not be deployed (or if it's deployed, it belongs to a different build mode).

In our use case, we use the channels for different brand builds of the same application, like "Yahoo" and "AOL" builds (just as an example). Even if we deploy a "Yahoo News 0.4.1beta1" build, we still want it to update from the "yahoo" channel, not the "aol" channel, nor the "latest" channel.

Comment (beyondkmp) [reply packages/electron-updater/src/providers/GitHubProvider.ts]: Fixed

Comment (benbucksch) [reply packages/electron-updater/src/providers/GitHubProvider.ts]: @beyondkmp Thank you! :smile:

**Review by benbucksch — APPROVED** [review]

File: packages/electron-updater/src/providers/GitHubProvider.ts (hunk 1 above)
Comment (benbucksch) [reply packages/electron-updater/src/providers/GitHubProvider.ts]: (I don't see the button for resolving this conversation. But it's resolved for me.)

**Pull request merged**